esrally/client/factory.py

```diff
@@ -346,7 +347,16 @@ def cluster_distribution_version(hosts, client_options, client_factory=EsClientF
     # version number does not exist for serverless
     version_number = version.get("number", version_build_flavor)
 
-    return version_build_flavor, version_number, version_build_hash
+    serverless_operator = False
+    if version_build_flavor == "serverless":
```

Comment:
Can we use the `is_serverless` helper function for this?
https://github.com/elastic/rally/blob/b0b5e05a40552bc0a6a0d2f2191b95c56327c23d/esrally/utils/versions.py#L36-L37